diffrax/_integrate.py

```diff
@@ -162,32 +163,37 @@ def _check(term_cls, term, term_contr_kwargs, yi):
                 pass
             elif n_term_args == 2:
                 vf_type_expected, control_type_expected = term_args
-                vf_type = eqx.filter_eval_shape(term.vf, 0.0, yi, args)
+                try:
+                    vf_type = eqx.filter_eval_shape(term.vf, 0.0, yi, args)
+                except Exception as e:
+                    raise ValueError(f"Error while tracing {term}.vf: " + str(e))
                 vf_type_compatible = eqx.filter_eval_shape(
                     better_isinstance, vf_type, vf_type_expected
                 )
                 if not vf_type_compatible:
-                    raise ValueError
+                    raise ValueError(f"Vector field term {term} is incompatible.")
 
                 contr = ft.partial(term.contr, **term_contr_kwargs)
                 # Work around https://github.com/google/jax/issues/21825
-                control_type = eqx.filter_eval_shape(contr, 0.0, 0.0)
+                try:
+                    control_type = eqx.filter_eval_shape(contr, 0.0, 0.0)
+                except Exception as e:
+                    raise ValueError(f"Error while tracing {term}.contr: " + str(e))
                 control_type_compatible = eqx.filter_eval_shape(
                     better_isinstance, control_type, control_type_expected
                 )
                 if not control_type_compatible:
-                    raise ValueError
+                    raise ValueError(f"Control term {term} is incompatible.")
             else:
                 assert False, "Malformed term structure"
             # If we've got to this point then the term is compatible
 
     try:
         with jax.numpy_dtype_promotion("standard"):
             jtu.tree_map(_check, term_structure, terms, contr_kwargs, y)
-    except ValueError:
+    except Exception as e:
         # ValueError may also arise from mismatched tree structures
-        return False
-    return True
+        raise ValueError("Terms are not compatible with solver! " + str(e))
```

Comment:
I think this needs either a `from e` after the end, to chain together exceptions.

Reply:
Ah yes, the beauty of python 3